data_structures/kd_tree/tests/test_kdtree.py

```diff
@@ -8,46 +8,41 @@
 
 
 @pytest.mark.parametrize(
-    ("num_points", "cube_size", "num_dimensions", "depth", "expected_result"),
+    ("num_points", "cube_size", "num_dimensions", "expected_result"),
     [
-        (0, 10.0, 2, 0, None),  # Empty points list
-        (10, 10.0, 2, 2, KDNode),  # Depth = 2, 2D points
-        (10, 10.0, 3, -2, KDNode),  # Depth = -2, 3D points
+        (0, 10.0, 2, None),  # Empty points list
+        (10, 10.0, 2, KDNode),  # 2D points
+        (10, 10.0, 3, KDNode),  # 3D points
     ],
 )
-def test_build_kdtree(num_points, cube_size, num_dimensions, depth, expected_result):
+def test_build_kdtree(num_points, cube_size, num_dimensions, expected_result):
     """
     Test that KD-Tree is built correctly.
 
     Cases:
         - Empty points list.
-        - Positive depth value.
-        - Negative depth value.
+        - Valid points list with correct dimensions.
     """
     points = (
         hypercube_points(num_points, cube_size, num_dimensions).tolist()
         if num_points > 0
         else []
     )
 
-    kdtree = build_kdtree(points, depth=depth)
+    kdtree = build_kdtree(points)
```

Comment:
you are always calling `build_kdtree()` with the default case `depth = 0`. This is not going to test it correctly.